File: app/src/main/java/com/femtioprocent/omega/util/GenericEventManager.java
package com.femtioprocent.omega.util;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

public class GenericEventManager {
    List<GenericEventListener> li = new ArrayList<>();

    public void fireGenericEvent(GenericEvent ge, Object a) {
        Iterator<GenericEventListener> it = li.iterator();
        while (it.hasNext()) {
            GenericEventListener gel = (GenericEventListener) it.next();
            if (true /*gel.grp.equals(ge.grp) */)
                gel.genericEvent(ge, a);
        }
    }

    public void addGenericEventListener(GenericEventListener gel) {
        li.add(gel);
    }
}

File: app/src/main/kotlin/com/femtioprocent/omega/util/GenericEventManager.kt
package com.femtioprocent.omega.util

class GenericEventManager {
    var li: MutableList<GenericEventListener> = ArrayList()
    fun fireGenericEvent(ge: GenericEvent?, a: Any?) {
	li.forEach {if (true /*gel.grp.equals(ge.grp) */) it.genericEvent(ge, a) }
    }

    fun addGenericEventListener(gel: GenericEventListener) {
	li.add(gel)
    }
}
